### PR TITLE
Feature/dashboard program

### PR DESCRIPTION
**Pull request by overnin**
- [x] Please feedback here on the UX

### CODE REVIEW

**Comment by markphi2013** [comment]
the graphs cannot load cause of the errors in the screen shot below 

screenshot from 2015-10-20 07 36 35

**Comment by markphi2013** [comment]
When the screen is minimized to 900px the " View schedule item(s) over next " graph is cut on the right side and user can't view some info

screenshot from 2015-10-20 07 40 17

**Comment by overnin** [comment]
@markphi2013 i fixed the depencies, does it load?

**Comment by markphi2013** [comment]
not loading still

screenshot from 2015-10-20 15 21 05

**Comment by markphi2013** [comment]
- [ ] Schedule is not loading
- [ ] the there is big row spacing on row zero 

capture

**Comment by markphi2013** [comment]
- [ ] No X axis line on the graphs  
capture2

**Comment by markphi2013** [comment]
- [ ]  it would be nice to have export or pop up the dashboard view  to view only the graphs on a separate window 

- [ ] make the graphs more responsive to screen size

**Comment by overnin** [comment]
@markphi2013 added the most active receivers... not yet clear how to design the layout, suggestion welcome.

**Comment by markphi2013** [comment]
when i have so many keywords the layout is changed shown in the image below 
screenshot from 2015-10-22 08 40 41

About the design layout 
the hits value should may be have different color like the lines in the graphs 

```
Most active message receivers:
Dialogues:                       Requests:
          7 - 2040                         4 - ft
                                           5 - t1
                                           6 - t2
                                                 
```

**Comment by markphi2013** [comment]
- [ ]  "restart worker" button is missing

**Comment by overnin** [comment]
Good point but i would move it to another screen. I identified the
following options:
- program edit page
- settings edit page
- create a new page in the administration section.
What do u think?

**Comment by overnin** [comment]
@markphi2013 ready for feedback

**Comment by markphi2013** [comment]
backend tests not passing and backend not running 

```
(ve)vagrant@vusion:/var/vusion/backend$ trial vusion
vusion ...                                                              [ERROR]

===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/var/vusion/backend/ve/local/lib/python2.7/site-packages/twisted/trial/runner.py", line 586, in loadByNames
    things.append(self.findByName(name))
  File "/var/vusion/backend/ve/local/lib/python2.7/site-packages/twisted/trial/runner.py", line 395, in findByName
    return filenameToModule(name)
  File "/var/vusion/backend/ve/local/lib/python2.7/site-packages/twisted/trial/runner.py", line 84, in filenameToModule
    ret = reflect.namedAny(reflect.filenameToModuleName(fn))
  File "/var/vusion/backend/ve/local/lib/python2.7/site-packages/twisted/python/reflect.py", line 303, in namedAny
    topLevelPackage = _importAndCheckStack(trialname)
  File "/var/vusion/backend/ve/local/lib/python2.7/site-packages/twisted/python/reflect.py", line 250, in _importAndCheckStack
    reraise(excValue, excTraceback)
  File "/var/vusion/backend/vusion/__init__.py", line 1, in <module>
    from dialogue_worker import DialogueWorker
  File "/var/vusion/backend/vusion/dialogue_worker.py", line 34, in <module>
    from vusion.component import (
  File "/var/vusion/backend/vusion/component/__init__.py", line 1, in <module>
    from dialogue_worker_property_helper import DialogueWorkerPropertyHelper
  File "/var/vusion/backend/vusion/component/dialogue_worker_property_helper.py", line 12, in <module>
    from vusion.persist import Shortcode
  File "/var/vusion/backend/vusion/persist/__init__.py", line 55, in <module>
    from program_setting.program_setting_manager import ProgramSettingManager
exceptions.ImportError: No module named program_setting.program_setting_manager

vusion
-------------------------------------------------------------------------------
Ran 1 tests in 0.005s

```

**Comment by overnin** [comment]
it was a missing file, @markphi2013  please review

**Comment by markphi2013** [comment]
Backend failing tests

```
[FAIL]
Traceback (most recent call last):
  File "/var/vusion/backend/ve/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 1099, in _inlineCallbacks
    result = g.send(result)
  File "/var/vusion/backend/vusion/tests/test_dialogue_worker_consume_control.py", line 111, in test_consume_control_mass_tag_empty_selector
    self.assertEqual(1, self.collections['schedules'].count())
  File "/var/vusion/backend/ve/local/lib/python2.7/site-packages/twisted/trial/_synctest.py", line 356, in assertEqual
    % (msg, pformat(first), pformat(second)))
twisted.trial.unittest.FailTest: not equal:
a = 1
b = 0


vusion.tests.test_dialogue_worker_consume_control.DialogueWorkerTestCase_consumeControlMessage.test_consume_control_mass_tag_empty_selector
===============================================================================
[FAIL]
Traceback (most recent call last):
  File "/var/vusion/backend/ve/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 1099, in _inlineCallbacks
    result = g.send(result)
  File "/var/vusion/backend/vusion/tests/test_dialogue_worker_send_schedule.py", line 70, in test_send_scheduled_messages
    self.assertEqual(len(messages), 3)
  File "/var/vusion/backend/ve/local/lib/python2.7/site-packages/twisted/trial/_synctest.py", line 356, in assertEqual
    % (msg, pformat(first), pformat(second)))
twisted.trial.unittest.FailTest: not equal:
a = 4
b = 3


vusion.tests.test_dialogue_worker_send_schedule.DialogueWorkerTestCase_sendSchedule.test_send_scheduled_messages
===============================================================================
[FAIL]
Traceback (most recent call last):
  File "/var/vusion/backend/ve/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 1099, in _inlineCallbacks
    result = g.send(result)
  File "/var/vusion/backend/vusion/tests/test_export_worker.py", line 351, in test_export_failed_exception
    self.assertEqual(export['failure-reason'], 'No such file or directory')
  File "/var/vusion/backend/ve/local/lib/python2.7/site-packages/twisted/trial/_synctest.py", line 356, in assertEqual
    % (msg, pformat(first), pformat(second)))
twisted.trial.unittest.FailTest: not equal:
a = u"command SON([('aggregate', u'exports'), ('pipeline', [{'$match': {'status': 'success'}}, {'$group': {'total': {'$sum': '$size'}, '_id': None}}]), ('useCursor', False)]) failed: unrecognized field 'useCursor'"
b = 'No such file or directory'


vusion.tests.test_export_worker.ExportWorkerTestCase.test_export_failed_exception
===============================================================================
[FAIL]
Traceback (most recent call last):
  File "/var/vusion/backend/ve/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 1099, in _inlineCallbacks
    result = g.send(result)
  File "/var/vusion/backend/vusion/tests/test_export_worker.py", line 334, in test_export_failed_max_total_export_size
    self.assertEqual(export['status'], 'no-space')
  File "/var/vusion/backend/ve/local/lib/python2.7/site-packages/twisted/trial/_synctest.py", line 356, in assertEqual
    % (msg, pformat(first), pformat(second)))
twisted.trial.unittest.FailTest: not equal:
a = u'failed'
b = 'no-space'


vusion.tests.test_export_worker.ExportWorkerTestCase.test_export_failed_max_total_export_size
===============================================================================
[FAIL]
Traceback (most recent call last):
  File "/var/vusion/backend/ve/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 1099, in _inlineCallbacks
    result = g.send(result)
  File "/var/vusion/backend/vusion/tests/test_export_worker.py", line 259, in test_export_history
    self.assertTrue(os.path.isfile('testing_export.csv'))
  File "/var/vusion/backend/ve/local/lib/python2.7/site-packages/twisted/trial/_synctest.py", line 308, in assertTrue
    raise self.failureException(msg)
twisted.trial.unittest.FailTest: None

vusion.tests.test_export_worker.ExportWorkerTestCase.test_export_history
===============================================================================
[FAIL]
Traceback (most recent call last):
  File "/var/vusion/backend/ve/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 1099, in _inlineCallbacks
    result = g.send(result)
  File "/var/vusion/backend/vusion/tests/test_export_worker.py", line 165, in test_export_participant_ordered_columns
    self.assertTrue(os.path.isfile('testing_export.csv'))
  File "/var/vusion/backend/ve/local/lib/python2.7/site-packages/twisted/trial/_synctest.py", line 308, in assertTrue
    raise self.failureException(msg)
twisted.trial.unittest.FailTest: None

vusion.tests.test_export_worker.ExportWorkerTestCase.test_export_participant_ordered_columns
===============================================================================
[FAIL]
Traceback (most recent call last):
  File "/var/vusion/backend/ve/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 1099, in _inlineCallbacks
    result = g.send(result)
  File "/var/vusion/backend/vusion/tests/test_export_worker.py", line 227, in test_export_participant_with_fakejoin
    self.assertTrue(os.path.isfile('testing_export.csv'))
  File "/var/vusion/backend/ve/local/lib/python2.7/site-packages/twisted/trial/_synctest.py", line 308, in assertTrue
    raise self.failureException(msg)
twisted.trial.unittest.FailTest: None

vusion.tests.test_export_worker.ExportWorkerTestCase.test_export_participant_with_fakejoin
===============================================================================
[FAIL]
Traceback (most recent call last):
  File "/var/vusion/backend/ve/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 1099, in _inlineCallbacks
    result = g.send(result)
  File "/var/vusion/backend/vusion/tests/test_export_worker.py", line 113, in test_export_participants_no_conditions
    self.assertTrue(os.path.isfile('testing_export.csv'))
  File "/var/vusion/backend/ve/local/lib/python2.7/site-packages/twisted/trial/_synctest.py", line 308, in assertTrue
    raise self.failureException(msg)
twisted.trial.unittest.FailTest: None

vusion.tests.test_export_worker.ExportWorkerTestCase.test_export_participants_no_conditions
===============================================================================
[FAIL]
Traceback (most recent call last):
  File "/var/vusion/backend/ve/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 1099, in _inlineCallbacks
    result = g.send(result)
  File "/var/vusion/backend/vusion/tests/test_export_worker.py", line 75, in test_export_participants_with_conditions
    self.assertTrue(os.path.isfile('testing_export.csv'))
  File "/var/vusion/backend/ve/local/lib/python2.7/site-packages/twisted/trial/_synctest.py", line 308, in assertTrue
    raise self.failureException(msg)
twisted.trial.unittest.FailTest: None

vusion.tests.test_export_worker.ExportWorkerTestCase.test_export_participants_with_conditions
===============================================================================
[FAIL]
Traceback (most recent call last):
  File "/var/vusion/backend/ve/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 1099, in _inlineCallbacks
    result = g.send(result)
  File "/var/vusion/backend/vusion/tests/test_export_worker.py", line 298, in test_export_unmatchable_reply
    self.assertEqual(export['status'], 'success')
  File "/var/vusion/backend/ve/local/lib/python2.7/site-packages/twisted/trial/_synctest.py", line 356, in assertEqual
    % (msg, pformat(first), pformat(second)))
twisted.trial.unittest.FailTest: not equal:
a = u'failed'
b = 'success'


vusion.tests.test_export_worker.ExportWorkerTestCase.test_export_unmatchable_reply
===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/var/vusion/backend/vusion/persist/export/tests/test_export_manager.py", line 56, in test_get_total_export_size
    self.manager.get_total_export_size(), 0)
  File "/var/vusion/backend/vusion/persist/export/export_manager.py", line 67, in get_total_export_size
    mongo_return = self.aggregate(pipeline, useCursor=False)
  File "/var/vusion/backend/vusion/persist/model_manager.py", line 49, in hooked
    result = orig_attr(*args, **kwargs)
  File "/var/vusion/backend/ve/local/lib/python2.7/site-packages/pymongo/collection.py", line 1336, in aggregate
    "aggregate", self.__name, **command_kwargs)
  File "/var/vusion/backend/ve/local/lib/python2.7/site-packages/pymongo/database.py", line 351, in _command
    msg, allowable_errors)
  File "/var/vusion/backend/ve/local/lib/python2.7/site-packages/pymongo/helpers.py", line 178, in _check_command_response
    raise OperationFailure(msg % errmsg, code, response)
pymongo.errors.OperationFailure: command SON([('aggregate', u'exports'), ('pipeline', [{'$match': {'status': 'success'}}, {'$group': {'total': {'$sum': '$size'}, '_id': None}}]), ('useCursor', False)]) failed: unrecognized field 'useCursor'

vusion.persist.export.tests.test_export_manager.TestExportManager.test_get_total_export_size
===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/var/vusion/backend/vusion/persist/history/tests/test_history_manager.py", line 525, in test_aggregate_count_per_day
    self.history_manager.aggregate_count_per_day()
  File "/var/vusion/backend/vusion/persist/history/history_manager.py", line 450, in aggregate_count_per_day
    cursor = self.collection.aggregate(pipeline, useCursor=True)
  File "/var/vusion/backend/ve/local/lib/python2.7/site-packages/pymongo/collection.py", line 1336, in aggregate
    "aggregate", self.__name, **command_kwargs)
  File "/var/vusion/backend/ve/local/lib/python2.7/site-packages/pymongo/database.py", line 351, in _command
    msg, allowable_errors)
  File "/var/vusion/backend/ve/local/lib/python2.7/site-packages/pymongo/helpers.py", line 178, in _check_command_response
    raise OperationFailure(msg % errmsg, code, response)
pymongo.errors.OperationFailure: command SON([('aggregate', u'history'), ('pipeline', [{'$match': {'message-direction': {'$exists': 1}}}, {'$group': {'count': {'$sum': 1}, '_id': {'direction': '$message-direction', 'month': {'$substr': ['$timestamp', 5, 2]}, 'day': {'$substr': ['$timestamp', 8, 2]}, 'year': {'$substr': ['$timestamp', 0, 4]}}}}, {'$project': {'count': 1, 'direction': '$_id.direction', '_id': {'$concat': ['$_id.year', '-', '$_id.month', '-', '$_id.day']}}}]), ('useCursor', True)]) failed: unrecognized field 'useCursor'

vusion.persist.history.tests.test_history_manager.TestHistoryManager.test_aggregate_count_per_day
-------------------------------------------------------------------------------
Ran 437 tests in 16.467s

FAILED (failures=10, errors=2, successes=425)
```

**Comment by overnin** [comment]
Please run again the install of pip requluirements

**Comment by markphi2013** [comment]
i think it's better with "View most receiving Dialogues and Request over post " and removing "over past" from the drop down

screenshot from 2015-11-11 10 37 31

**Comment by markphi2013** [comment]
The back end is running very well but no stats is being generated in the graphs

**Comment by overnin** [comment]
Mmmm do u see logs for the stats_worker ?

**Comment by overnin** [comment]
@markphi2013 the .yaml for the stat worker was missing, please pull and update the backend submodule.

**Comment by markphi2013** [comment]
- [ ] From graphs naming remove "View"
       <br/>  -- View messages to <b>"Messages"</b> 
       <br/>  --View most receiving Dialogues and Request to <b>"Most receiving Dialogues and Request over past"</b>
        <br/> --View scheduled item(s) over next to <b>"Scheduled item(s) over next"</b>

**Comment by markphi2013** [comment]
@overnin did some UI commits up for review . For me it's ok for merge

**Comment by overnin** [comment]
@markphi2013 do you agree on the following TODO list:
- [ ] having ```Most receiving ...``` and not with ```received```
- [ ] when looking a the last year for stats, don't use the day granularity by the month
- [ ] when selecting the ```since the beginning```, use month granularity
- [ ] when no results on ```Most receiving```, display the text ```no results over selected timeframe```.

File: app/Controller/ProgramHistoryController.php [hunk 1]
```diff
@@ -92,6 +93,61 @@ public function index()
         }
     }
 
+    /*
```
Comment (markphi2013) [inline app/Controller/ProgramHistoryController.php]: remove commented  code

File: app/Model/History.php [hunk 2]
```diff
@@ -256,7 +256,89 @@ public function addDialogueContent($histories, $dialoguesInteractionsContent)
         return $histories;
     }
     
-    
+    /*
```
Comment (markphi2013) [inline app/Model/History.php]: remove commented code

File: app/Model/Schedule.php [hunk 3]
```diff
@@ -217,6 +217,79 @@ public function summary($dateTime)
         
     }
     
+    /*
```
Comment (markphi2013) [inline app/Model/Schedule.php]: remove commented code

**Comment by markphi2013** [comment]
in ProgramSpecificMongoModel.php line 75 function not used and no test for the model

**Comment by markphi2013** [comment]
ok for merge

**Pull request merged**